packages/itwin/measure-tools/src/measurements/PerpendicularDistanceMeasurement.ts

```diff
@@ -0,0 +1,141 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Point3d } from "@itwin/core-geometry";
+import { ColorDef, LinePixels } from "@itwin/core-common";
+import type { DecorateContext } from "@itwin/core-frontend";
+import { GraphicType, IModelApp, QuantityType } from "@itwin/core-frontend";
+import { FormatterUtils } from "../api/FormatterUtils";
+import type { MeasurementWidgetData } from "../api/Measurement";
+import { MeasurementPropertyHelper } from "../api/MeasurementPropertyHelper";
+import { MeasureTools } from "../MeasureTools";
+import { DistanceMeasurement } from "./DistanceMeasurement";
+
+export class PerpendicularDistanceMeasurement extends DistanceMeasurement {
```

Comment:
You did not define a `MeasurementSerializer` for your measurement type. It will not be persisted nor restored properly.

Reply:
Added `PerpendicularDistanceMeasurementSerializer`